flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitor.java

```diff
@@ -188,9 +189,12 @@ public WebRuntimeMonitor(
 
 			// Cancel a job via GET (for proper integration with YARN this has to be performed via GET)
 			.GET("/jobs/:jobid/yarn-cancel", handler(new JobCancellationHandler()))
-			// DELETE is the preferred way of cancelling a job (Rest-conform)
+			// DELETE is the preferred way of canceling a job (Rest-conform)
 			.DELETE("/jobs/:jobid", handler(new JobCancellationHandler()))
 
+			// stop a job
+			.DELETE("/jobs/:jobid/stop", handler(new JobStoppingHandler()))
+
```

Comment:
`DELETE` is more conforming to the REST framework. `GET` is just a workaround to support yarn when working on the AM proxy page. Yarn doesn't transfer anything besides `GET` to the AM's http server.